diff --git a/android/build.gradle b/android/build.gradle
index 61ed6d6a840525f0b28f0c9924b1e2ea6db2f5c2..2656c46d34717ce961075da72efc0aaa673d202c 100644
--- a/android/build.gradle
+++ b/android/build.gradle	
@@ -77,6 +77,7 @@ android {
 
   namespace "expo.modules"
   defaultConfig {
+    minSdkVersion 24
     consumerProguardFiles 'proguard-rules.pro'
     versionCode 1
     versionName "3.0.29"
@@ -89,6 +90,7 @@ android {
       cmake {
         abiFilters(*reactNativeArchitectures())
         arguments "-DANDROID_STL=c++_shared",
+          "-DANDROID_PLATFORM=android-24",
           "-DANDROID_SUPPORT_FLEXIBLE_PAGE_SIZES=ON",
           "-DREACT_NATIVE_DIR=${expoModuleExtension.reactNativeDir}",
           "-DREACT_NATIVE_TARGET_VERSION=${expoModuleExtension.reactNativeVersion.minor}",
diff --git a/expo-module-gradle-plugin/src/main/kotlin/expo/modules/plugin/ProjectConfiguration.kt b/expo-module-gradle-plugin/src/main/kotlin/expo/modules/plugin/ProjectConfiguration.kt
index 7c2206904731d543dd3c59664edf2c4544fd835b..650a950347682ea61c9bb97b22257bbe019a2d74 100644
--- a/expo-module-gradle-plugin/src/main/kotlin/expo/modules/plugin/ProjectConfiguration.kt
+++ b/expo-module-gradle-plugin/src/main/kotlin/expo/modules/plugin/ProjectConfiguration.kt	
@@ -52,8 +52,7 @@ internal fun Project.applyDefaultAndroidSdkVersions() {
     applySDKVersions(
       compileSdk = rootProject.extra.safeGet("compileSdkVersion")
         ?: logger.warnIfNotDefined("compileSdkVersion", 36),
-      minSdk = rootProject.extra.safeGet("minSdkVersion")
-        ?: logger.warnIfNotDefined("minSdkVersion", 24),
+      minSdk = maxOf(24, (rootProject.extra.safeGet("minSdkVersion") as? Int) ?: 24),
       targetSdk = rootProject.extra.safeGet("targetSdkVersion")
         ?: logger.warnIfNotDefined("targetSdkVersion", 36)
     )
